AGTRAP.py

```diff
@@ -0,0 +1,244 @@
+import numpy as np
+import os
+from Bio import SeqIO
+from Bio.SeqUtils import gc_fraction
+from abc import ABC, abstractmethod
+from typing import Union
+
+
+def filter_read(input_path: str, output_filename: str = None,
+                gc_bounds: Union[tuple[Union[float, int]], Union[float, int]] = (0, 100),
+                length_bounds: Union[tuple[int], int] = (0, 2 ** 32),
+                quality_threshold: int = 0,
+                report: bool = False):
+
+    """
+    Filter the fastq file using the following parameters:
+    - GC content
+    - sequence length
+    - sequence quality
+
+    :param input_path: path to the fastq file that needs to be filtered
+    :param output_filename: file name with filtered data
+    :param gc_bounds: gc_bounds: gc content range to filtering (in percent);
+           sequences that are not included in the range are discarded;
+           if you pass one number, this number will be considered the upper limit;
+           default = (0, 100) - all reads will be saved, regardless of gc composition
+    :param length_bounds:sequence length range for filtering;
+           sequences that are not included in the range are discarded;
+           if you pass one number, this number will be considered the upper limit;
+           default = (0, 2 ** 32)
+    :param quality_threshold:average sequence quality threshold for filtering (Phred33 scale);
+           sequences with quality below the threshold are discarded
+    :param report:report = True (show filtering report), report = True (default)
+    :return:A filtered fastq file containing sequences
+             that have passed all tests (according to specified parameters)
+    """
+
+    if not isinstance(gc_bounds, tuple):
+        gc_bounds = (0, gc_bounds)
+    if not isinstance(length_bounds, tuple):
+        length_bounds = (0, length_bounds)
+    if output_filename is None:
+        output_filename = os.path.splitext(os.path.basename(input_path))[0]
+    output_filename = f'{output_filename}.fastq'
+
+    os.makedirs('fastq_filtrator_resuls', exist_ok=True)
+    output_dir = 'fastq_filtrator_resuls'
+
+    count_filtered_record = 0
+    count_record = 0
+
+    with open(os.path.join(output_dir, output_filename), mode='w') as out_file:
+        for record in SeqIO.parse("reads.3.fastq", "fastq"):

```

Comment:
Кажется кто-то забыл тут свой файлик. Возможно, должно быть 
```suggestion
        for record in SeqIO.parse(input_path, "fastq"):
```
:)

Reply:
Хе-хе, да